receiver/statsdreceiver/receiver.go

```diff
@@ -110,10 +122,13 @@ func (r *statsdReceiver) Start(ctx context.Context, _ component.Host) error {
 				batchMetrics := r.parser.GetMetrics()
 				for _, batch := range batchMetrics {
 					batchCtx := client.NewContext(ctx, batch.Info)
-
-					if err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer); err != nil {
+					numPoints := batch.Metrics.DataPointCount()
+					r.obsrecv.StartMetricsOp(batchCtx)
```

Comment:
This function returns a `context.Context` and thats the context that should be used in `Flush` and `EndMetricsOp`

Reply:
oops didn't understand the comment previously. updated again. PTAL